resources/authgear/templates/en/web/authflowv2/signup.html

```diff
@@ -50,38 +55,47 @@ <h2 class="screen-description">
-            autocapitalize="none"
-            name="q_login_id"
-            placeholder="{{ template "v2-placeholder.phone" }}"
-            data-retain-form-form-target="input"
-            data-retain-form-form-name-param="phone"
-          >
-        {{ else }}
-          <!-- TODO(tung): Update to use text input (https://github.com/authgear/authgear-server/issues/3557) -->
-          <input
-            {{ if $.ShouldFocusInput }}autofocus{{ end }}
-            class="block w-full"
-            tabindex="0"
-            type="{{ $.NonPhoneLoginIDInputType }}"
-            name="q_login_id"
-            autocomplete="username"
-            autocapitalize="none"
-            placeholder="{{ template "v2-placeholder.login-id" (dict "variant" $.LoginIDKey) }}"
-            data-controller="retain-form-input"
-            data-action="input->retain-form-input#input retain-form-input:input->retain-form-form#input"
-            data-retain-form-form-target="input"
-            data-retain-form-form-name-param="text"
-          >
-        {{ end }}
+        <div data-controller="text-field">
+          {{ if eq .login_id_type "phone" }}
+            <!-- TODO(tung): Update to use phone input (https://github.com/authgear/authgear-server/issues/3561) -->
+            <input
+              data-text-field-target="input"
+              class="block input w-full"
+              tabindex="0"
+              {{ if $.ShouldFocusInput }}autofocus{{ end }}
+              type="text"
+              inputmode="tel"
+              autocomplete="tel-national"
+              autocapitalize="none"
+              name="q_login_id"
+              placeholder="{{ template "v2-placeholder.phone" }}"
+              data-retain-form-form-target="input"
+              data-retain-form-form-name-param="phone"
+            >
+          {{ else }}
+            <!-- TODO(tung): Update to use text input (https://github.com/authgear/authgear-server/issues/3557) -->
+            <input
+              data-text-field-target="input"
+              {{ if $.ShouldFocusInput }}autofocus{{ end }}
+              class="block input w-full {{ if $display_error }}input--error{{end}}"
+              tabindex="0"
+              type="{{ $.NonPhoneLoginIDInputType }}"
+              name="q_login_id"
+              autocomplete="username"
+              autocapitalize="none"
+              placeholder="{{ template "v2-placeholder.login-id" (dict "variant" $.LoginIDKey) }}"
+              data-controller="retain-form-input"
+              data-action="input->retain-form-input#input retain-form-input:input->retain-form-form#input"
+              data-retain-form-form-target="input"
+              data-retain-form-form-name-param="text"
+            >
+          {{ end }}
+          {{ if $display_error }}
+            <ul class="input__error-message mt-2" data-text-field-target="errorMessage">
+              {{ template "authflowv2/__error.html" $ }}
+            </ul>
```

Comment:
Not sure if we should use `ul` here. Because ultimately we will only display one message?

Reply:
@Ngkaokis @IniZio @louischan-oursky 
I think it depends on what __error.html do.
In this pr, it creates <li> elements, so I used `<ul>` here. The advantage is it will be able to handle multiple error messages. The disadvantage is there is an assumption that it must be used as html element.
We can change __error.html to render texts, so it can be used in attributes too. See if any opinion on it?

Reply:
How do you expect to handle form with multiple fields? For example, in create password screen, there are enter password field and confirm password field and both can display corresponding error message. Do you expect to have two `__error.html` under each field and conditionally render each one?

Reply:
> I think it depends on what __error.html do.
> In this pr, it creates `<li>` elements, so I used `<ul>` here. 

But it will be something look like the following eventually in the dom?
```
<ul>
<li>Error Message</li>
</ul>
```

The `<ul>` and `<li>` seems not necessary?
Seem just `<p>ErrorMessage</p>` is enough?

Reply:
@Ngkaokis I think there should only be one place to display the error message. For example, in create password screen, the error message should always be under the first text input.

However, if there are really cases we need two error messages, we need to conditionally render the template with a correct context.

Reply:
> I think there should only be one place to display the error message. For example, in create password screen, the error message should always be under the first text input.

Oh, I thought error like confirm password not the same as the above one would be displayed under the second one

Reply:
For the actual error position, I think we can discuss it on implementing create password screen.
Maybe let me change `__error.html` to render text only first which seems more flexible.